# ★Chapter4　Pythonの基礎

- **4.1 基礎**
    - **4.1.1 Hello world**
    - **4.1.2 Pythonの用途**
    - **4.1.3 コメントの入力**
    - **4.1.4 数値と文字列**
    - **4.1.5 演算**
<br><br>
- **4.2 変数**
    - **4.2.1 変数** 
    - **4.2.2 変数の更新**
    - **4.2.3 文字列の連結**
<br><br>
- **4.3 型**
    - **4.3.1 型**
    - **4.3.2 型の変換**
    - **4.3.3 型の理解と確認**
<br><br>
- **4.4 if文**
    - **4.4.1 比較演算子**
    - **4.4.2 if文**
    - **4.4.3 else**
    - **4.4.4 elif**
    - **4.4.5 and・not・or**
<br><br>
- **添削問題**

## ●4.1 基礎

### ○4.1.1 Hello world

#### □問題

In [ ]:
# Pythonでは行頭に#を置いた行はコメントして扱われます
# 「Hello world」と出力してください
print("Hello world")

**リスト 4.1：問題**

#### □解答例

In [1]:
# Pythonでは行頭に#を置いた行はコメントして扱われます
# 「Hello world」と出力してください
print("Hello world")

Hello world


**リスト 4.2：解答例**

### ○4.1.2  Python の用途

この項にサンプルはありません

### ○4.1.3 コメントの入力

#### □問題

In [ ]:
# 5 + 2の結果を出力してください
print(5 + 2)


print(3 + 8)


**リスト4.3：問題**

#### □解答例

In [3]:
# 5 + 2の結果を出力してください
print(5 + 2)

# 3 + 8の結果を出力する
print(3 + 8)

7
11


**リスト4.4：解答例**

### ○4.1.4 数値と文字列

In [4]:
print(3 + 6)

9


**リスト 4.5：計算の例①**

In [5]:
print("8 - 3")

8 - 3


**リスト4.6：代入の例②**

#### □問題

In [ ]:
# 数値の「18」を出力してください


# 数値の 2 に 6 を足したものを出力してください


#「2 + 6」という文字列を出力してください


**リスト4.7：問題**

#### □解答例

In [3]:
# 数値の「18」を出力してください
print(18)

# 数値の 2 に 6 を足したものを出力してください
print(2 + 6)

# 「2 + 6」という文字列を出力してください
print("2 + 6")

18
8
2 + 6


**リスト4.8：解答例**リスト番号とる

### ○4.1.5 演算

#### □問題

In [ ]:
# 3 + 5


# 3 - 5


# 3 × 5


# 3 ÷ 5


# 3を5で割った余り


# 3の5乗



**リスト4.9：問題**

#### □解答例

In [4]:
# 3 + 5
print(3 + 5)

# 3 - 5
print(3 - 5)

# 3 × 5
print(3 * 5)

#　3 ÷ 5
print(3 / 5)

# 3を5で割った余り
print(3 % 5)

# 3の5乗
print(3 ** 5)


8
-2
15
0.6
3
243


**リスト4.10：解答例**

## ●4.2 変数


### ○4.2.1 変数

In [5]:
n = "いぬ"
print(n)

いぬ


**リスト4.11：変数の例①**

In [ ]:
# printを変数名に使うと、print()を呼び出す段階でエラーが出ます
print = "Hello"
print(print) # TypeError: 'str' object is not callable

**リスト4.12：変数の例②**

#### □問題

In [ ]:
# 変数nに「ねこ」を代入してください


# 変数nを出力してください


# 「n」という文字列を出力してください


# 変数nに3 + 7という数式を代入してください


# 変数nを出力してください


**リスト4.13：問題**

#### 解答例

In [6]:
# 変数nに「ねこ」を代入してください
n = "ねこ"

# 変数nを出力してください
print(n)

# 「n」という文字列を出力してください。
print("n")

# 変数nに3 + 7という数式を代入してください
n = 3 + 7

# 変数nを出力してください
print(n)

ねこ
n
10


**リスト4.14：解答例**

### 〇4.2.2 変数の更新

In [7]:
x = 1
print(x)  
x = x + 1
print(x)  

1
2


**リスト4.15：変数の更新の例①**

In [8]:
x = 5
x *= 2
print(x)  

10


**リスト4.16：変数の更新の例②**

#### □問題

In [ ]:
m = "ねこ"
print(m)

# 変数mに「いぬ」を上書きして、出力してください




n = 14
print(n)

# 変数nに5を掛けて、上書きしてください


print(n)

**リスト4.17：問題**

#### □解答例

In [1]:
m = "ねこ"
print(m)

# 変数mに「いぬ」を上書きして、出力してください
m = "いぬ"
print(m)


n = 14
print(n)

# 変数nに5を掛けて、上書きしてください。n = n * 5、n = 5 * n　も正解です
n *= 5

print(n)

ねこ
いぬ
14
70


**リスト4.18：解答例**

### 〇4.2.3 文字列の連結

In [10]:
m = "太郎"
print("私の名前は" + m + "です") 

私の名前は太郎です


**リスト4.19：⽂字列の連結の例**

#### □問題

In [ ]:
# 変数pに「東京」を代入してください


# 変数pを用いて「私は東京出身です」と出力してください



**リスト4.20：問題**

 #### □解答例

In [2]:
# 変数pに「東京」を代入してください
p = "東京"

# 変数pを用いて「私は東京出身です」と出力してください
print("私は" + p + "出身です")

私は東京出身です


**リスト4.21：解答例**

## ●4.3 型

### ○4.3.1 型

In [ ]:
height = 177
print("身長は" + height + "cmです。")   # 「TypeError: must be str, not int」というエラーが出力されます

**リスト 4.22：型の例①**

In [1]:
height = 177
type(height)  # int 型であることがわかります

int

**リスト 4.23：型の例②**

#### □問題

In [ ]:
h = 1.7
w = 60

# 変数h,wの型を出力してください



# 変数bmiに計算結果を代入してください


# 変数bmiを出力してください


# 変数bmiの型を出力してください


**リスト4.24：問題**

#### □解答例

In [13]:
h = 1.7
w = 60

# 変数h,wの型を出力してください
print(type(h))
print(type(w))

# 変数bmiに計算結果を代入してください
bmi = w / h ** 2

# 変数bmiを出力してください
print(bmi)

# 変数bmiの型を出力してください
print(type(bmi))

<class 'float'>
<class 'int'>
20.761245674740486
<class 'float'>


**リスト4.25：解答例**

 ### ○4.3.2 型の変換

In [17]:
h = 177
print("身長は" + str(h) + "cmです")

身長は177cmです


**リスト4.26：型の変換の例①**

In [18]:
a = 35.4
b = 10
print(a + b)

45.4


**リスト4.27：型の変換の例②**

#### □問題

In [ ]:
h = 1.7
w = 60
bmi = w / h ** 2

# 「あなたのbmiは〇〇です」と出力してください
print("あなたのbmiは" + bmi + "です")

**リスト 4.28：問題**

#### □解答例

In [14]:
h = 1.7
w = 60
bmi = w / h ** 2

# 「あなたのbmiは〇〇です」と出力してください
print("あなたのbmiは" + str(bmi) + "です")

あなたのbmiは20.761245674740486です


**リスト4.29：解答例**

### ○4.3.3 型の理解確認

In [15]:
greeting = "yo!"
print(greeting*2)  

yo!yo!


**リスト4.30：型の理解の確認の例**

#### □問題

In [19]:
n = "10"
print(n*3)

101010


**リスト 4.31：問題**

- int型で30
- int型で101010
- str型で30
- str型で101010

#### □解答例

str型で101010

## ●4.4 if文

### ○4.4.1 比較演算子

In [19]:
print(1 + 1 == 3)  

False


**リスト4.32：⽐較演算⼦の例**

#### □問題

In [26]:
# 4 + 6 と -10 を「!=」を用いて関係式を作り、True を出力してください



**リスト4.33：問題**

#### □解答例

In [20]:
# 4 + 6 と -10 を「!=」を用いて関係式を作り、True を出力してください
print(4 + 6 != -10)

True


**リスト4.34：解答例**

### ○4.4.2 if文

In [22]:
n = 2
if n == 2:
    print("残念！あなたは" + str(n) + "番目の到着です")  # nが2の時だけ表示されます

残念！あなたは2番目の到着です


**リスト4.35：if ⽂の例①**

In [20]:
animal = "cat"
if animal == "cat":
    print("ねこはかわいいですな")  # animalがcatの時だけ表示されます

ねこはかわいいですな


**リスト 4.36：if ⽂の例②**

#### □問題

In [30]:
n = 16

# ifを用いて変数 n が 15 より大きい場合「とても大きい数字」と出力してください


**リスト 4.37：問題**

#### □解答例

In [23]:
n = 16

# ifを用いて変数 n が 15 より大きい場合「とても大きい数字」と出力してください
if n > 15:
    print("とても大きい数字")

とても大きい数字


**リスト4.38：解答例**

### ○4.4.3 else

In [27]:
n = 2
if n == 1:
    print("優勝おめでとう！")  # nが1の時だけ、表示されます
else:
    print("残念！あなたは" + str(n) + "番目の到着です")  # nが1ではない時、表示されます

残念！あなたは2番目の到着です


**リスト4.39：elseの例①**

In [25]:
animal = "cat"
if animal == "cat":
    print("ねこはかわいいですな")  # animalが "cat"の時だけ、表示されます
else:
    print("これはねこじゃないにゃ")  # animalが "cat"ではない時、表示されます

ねこはかわいいですな


**リスト4.40：else の例②**

#### □問題

In [ ]:
n = 14


if n > 15:
    print("とても大きい数字。")
# elseを用いて「小さい数字」を出力してください


**リスト4.41：問題**

#### □解答例

In [8]:
n = 14


if n > 15:
    print("とても大きい数字。")
# elseを用いて「小さい数字」を出力してください
else:
    print("小さい数字")

小さい数字


**リスト4.42：解答例**

### ○4.4.4 elif

In [9]:
number = 2
if number == 1:
    print("金メダルです！")
elif number == 2:
    print("銀メダルです！")
elif number == 3:
    print("銅メダルです！")
else:
    print("残念！あなたは" + str(number) + "番目の到着です。") 

銀メダルです！


**リスト4.43：elif の例①**

In [10]:
animal = "cat"
if animal == "cat":
    print("ねこはかわいいですな")
elif animal == "dog":
    print("犬はかっこいいわん")
elif animal == "elephant":
    print("象はおおきいぞう")
else:
    print("ねこでも犬でも象でもないにゃん")

ねこはかわいいですな


**リスト4.44：elif の例②**

#### □問題

In [ ]:
n = 14

if n > 15:
    print("とても大きい数字")
# elifを用いて n が 11 以上 15 以下の時、「中くらいの数字」と出力してください


    
else:
    print("小さい数字")

**リスト4.45：問題**

#### □解答例

In [29]:
n = 14

if n > 15:
    print("とても大きい数字")
# elifを用いて n が 11 以上 15 以下の時、「中くらいの数字」と出力してください
elif n >= 11:
    print("中くらいの数字")
else:
    print("小さい数字")

中くらいの数字


**リスト4.46：解答例**

### ○4.4.5 and・not・or

#### □問題

In [38]:
n_1 = 14
n_2 = 28

# n_1が 8 より大きく 14 より小さい条件式を作り andを用いて出力してください
print()

# n_1の 2 乗が n_2の 5 倍より小さい条件式を作り notを用いて出力してください
print()


**リスト4.47：問題**

#### □解答例

In [28]:
n_1 = 14
n_2 = 28

# n_1が 8 より大きく 14 より小さい条件式を作り andを用いて出力してください
print(n_1 > 8 and n_1 < 14)

# n_1の 2 乗が n_2の 5 倍より小さい条件式を作り notを用いて出力してください
print(not n_1 ** 2 < n_2 * 5)

False
True


**リスト4.48：解答例**

## ●4.5 添削問題

#### □問題

In [ ]:
# 変数yearに西暦で年を入力してください





# if文により条件分岐をし、うるう年か平年かを判別します


**リスト4.49：問題**

#### □解答例

In [32]:
# 変数yearに西暦で年を入力してください
year = 2000

# if文により条件分岐をし、うるう年か平年かを判別します
if year % 100 == 0 and year % 400 != 0:
    print (str(year) + "年は平年です")
elif year % 4 == 0:
    print (str(year) + "年はうるう年です")
else:
    print (str(year) + "年は平年です")

2000年はうるう年です


**リスト4.50：解答例**